fhir-path/src/main/java/com/ibm/fhir/path/evaluator/FHIRPathEvaluator.java

```diff
@@ -1457,5 +1462,59 @@ public void unsetExternalConstant(String name) {
         public boolean hasExternalConstant(String name) {
             return externalConstantMap.containsKey(name);
         }
+
+        /**
+         * Set the constraint currently under evaluation
+         *
+         * @param constraint
+         *     the constraint currently under evaluation
+         */
+        public void setConstraint(Constraint constraint) {
+            this.constraint = constraint;
+        }
+
+        /**
+         * Unset the constraint currently under evaluation
+         */
+        public void unsetConstraint() {
```

Comment:
when is unsetting the constraint required? when is it not required?

Reply:
The unset is driven by the Validator (after it has evaluated the current constraint and before it changes to the context constraint).